### DISCUSSION TITLE
car/cdr procedures?

### DISCUSSION THREAD

**Discussion by furesoft** [rzubek/CSLisp#16] [Ideas]
what do you think about car/cdr procedures. may 

(car myprocedure) -> returns the name of the procedure or the package where it lives
(cdr myprocedure) -> gives the body of the procedure

**Comment by rzubek** [comment]
Since `car/cdr` are traditional names for cons accessors ( `first/rest` in our codebase) I would rather not overload them with a different semantics.

Maybe `(get-function-name myprocedure)` and `(get-function-body myprocedure)`?

Also - both of these will run into problems in that a procedure doesn't know its name. It only consists of a closure (lambda + pointer to parent environment). There could be many symbols in different packages pointing to the same closure (globals or local vars).

Is there a use case for doing an exhaustive search through all accessible environments and packages, looking for all the symbols that point to a specific closure?
